src/transformers/modeling_layoutlm.py

```diff
@@ -0,0 +1,472 @@
+# coding=utf-8
+# Copyright 2018 The Microsoft Research Asia LayoutLM Team Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch LayoutLM model. """
+
+
+import logging
+
+import torch
+from torch import nn
+from torch.nn import CrossEntropyLoss
+
+from .configuration_layoutlm import LayoutLMConfig
+from .file_utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_callable
+from .modeling_bert import BertLayerNorm, BertModel, BertOnlyMLMHead, BertPreTrainedModel
```

Comment:
The modeling files should be independent from each other for easy customization when researchers use them. Please copy all relevant classes from the `bert_modeling` file and paste them here.

Reply:
I think it will bring some redundant code. And @JetRunner said to me I can remain it unchanged.

Reply:
Yeah @sgugger I think our internal discussion reaches a consensus that we can allow some import if the code is exactly the same? As long as no inheritance?

Reply:
No, our internal discussions which resulted in the current [README](https://github.com/huggingface/transformers#why-should-i-use-transformers) (see point 4) are that we should copy and paste code so that our models are self-contained and easy to tweak to the cost of redundancy. We are aware it brings a cost to maintenance of the code but we are willing to pay it so that researchers can easily customize our models to their needs :-) There have been some instances in the past where the code is not properly duplicated that we will fix (RoBERTa for example) but we won't do the same mistake for new models.

The only inheritance we accept is for things like `Flaubert` or `XLM-RoBERTa` when the models are exactly the same as another model of the library (and we just have new classes for the proper names).

Reply:
Fair enough! I must have missed something. Anyway, @liminghao1630 maybe you should copy those classes to your modeling files? Thanks!

Reply:
OK, should I rename those classes or only copy them to our modeling file?

Reply:
Yeah, please rename BertXxx to LayoutLMXxx